experimental/ragged_inference/triton_v2_matmul.py

```diff
@@ -0,0 +1,196 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import torch
+import triton
+import triton.language as tl
+from triton.ops.matmul_perf_model import estimate_matmul_time, prune_num_stages
+
+# Credits: this comes directly from the Triton repo, authors are Da Yan and Phil Tillet
+# See https://github.com/openai/triton/blob/v2.0/python/triton/ops/matmul.py
+# copied here to help with development of new features, to be cleaned up
+
+
+def init_to_zero(name):
+    return lambda nargs: nargs[name].zero_()
+
+
+def get_configs_io_bound():
+    configs = []
+    for num_stages in [2, 3, 4, 5, 6]:
+        for block_m in [16, 32]:
+            for block_k in [32, 64]:
+                for block_n in [32, 64, 128, 256]:
+                    num_warps = 2 if block_n <= 64 else 4
+                    configs.append(
+                        triton.Config(
+                            {
+                                "BLOCK_M": block_m,
+                                "BLOCK_N": block_n,
+                                "BLOCK_K": block_k,
+                            },
+                            num_stages=num_stages,
+                            num_warps=num_warps,
+                        )
+                    )
+
+    return configs
+
+
+def get_all_configs():
+    return [
+        # basic configs for compute-bound matmuls
+        triton.Config(
```

Comment:
Where do these magic numbers / configs come from?

Reply:
comes from here, empirical good values for Ampere GPUs, architecture dependent but Triton does navigate around some specifics thanks for all these scheduling options https://github.com/openai/triton/blob/v2.0/python/triton/ops/matmul.py#L35